src/backends/oracle/statement.cpp

```diff
@@ -164,21 +164,29 @@ std::string oracle_statement_backend::rewrite_for_procedure_call(
 
 int oracle_statement_backend::prepare_for_describe()
 {
-    sword res = OCIStmtExecute(session_.svchp_, stmtp_, session_.errhp_,
-        1, 0, 0, 0, OCI_DESCRIBE_ONLY);
-    if (res != OCI_SUCCESS)
-    {
-        throw_oracle_soci_error(res, session_.errhp_);
-    }
+    ub2 statementType = 0;
+    sword res = OCIAttrGet(static_cast<dvoid*>(stmtp_),
+        static_cast<ub4>(OCI_HTYPE_STMT), static_cast<dvoid*>(&statementType),
+        0, static_cast<ub4>(OCI_ATTR_STMT_TYPE), session_.errhp_);
+
+    int cols = 0;
+    if (statementType == OCI_STMT_SELECT) {
```

Comment:
Minor style nitpick:
```suggestion
    if (statementType == OCI_STMT_SELECT)
    {
```